core/libp2p/multi/multistream.hpp

```diff
@@ -6,11 +6,66 @@
 #ifndef KAGOME_MULTISTREAM_HPP
 #define KAGOME_MULTISTREAM_HPP
 
+#include <filesystem>
+#include <gsl/span>
+
+#include "common/buffer.hpp"
+#include "common/result.hpp"
+
 namespace libp2p::multi {
   /**
    * Format of stream identifier used in Libp2p
+   * @see https://github.com/multiformats/multistream
    */
-  class Multistream {};
+  class Multistream {
+   public:
+    /**
+     * A protocol used in the stream is represented as a UNIX URI instead of just its name,
+     * since it's much more descriptive
+     */
+    using Path = std::filesystem::path;
+
+    /**
+     * Creates a Multistream object from
+     * a URI, which contains info about the protocol of the stream,
+     * @example /http/w3id.org/ipfs/1.1.0
+     * and a binary buffer with the stream content.
+     * @returns a Result object that contains either the Multistream, or an error, if any occured
+     */
+    static auto create(Path codecPath, const kagome::common::Buffer &data)
+        -> kagome::expected::Result<Multistream, std::string>;
+
+    /**
+     * Creates a Multistream object from
+     * a buffer with bytes representing a Multistream.
+     * <varint-length>'/'<codec-path>'\n'<data>
+     * @returns a Result object that contains either the Multistream, or an error, if any occured
+     */
+    static auto createFromBuffer(const kagome::common::Buffer &bytes)
+        -> kagome::expected::Result<Multistream, std::string>;
+
+    /**
+     * @return the URI with information about the protocol that is used in the stream
+     */
+    auto getCodecPath() const -> const Path &;
+
+    /**
+     * @return the content of the stream
+     */
+    auto getEncodedData() const -> gsl::span<const uint8_t>;
+
+    /**
+     * @return the buffer that contains the whole multistream
+     */
+    auto getBuffer() const -> const kagome::common::Buffer &;
+
```

Comment:
Please, also add methods `addPrefix` and `rmPrefix`, as they are specified in docs, and thus can be useful

Reply:
And also `getHex()` and `toString()`

Reply:
Why have you decided not to add those methods? They are in docs

Reply:
Do you mean JS example in the docs? I  thought that addPrefix and rmPrefix would not be much useful in this implementation because they are pretty much equivalent to re-creation of the multistream with an extended path, but now I actually can see why it would be useful. I will add them.